File: golang-io/fileCheck/fileCheck.go
package fileCheck

import (
	"io/ioutil"
)

func FileCheck(filePath string, dataBase *[][]string) {
	liste := []string{}
	files, _ := ioutil.ReadDir(filePath)

	for _, f := range files {
		if f.IsDir() == false {
			liste = append(liste, f.Name())
		} else {
			newPath := filePath + "/" + f.Name()
			FileCheck(newPath, dataBase)
		}
	}

	*dataBase = append(*dataBase, liste)
}

File: python-os/fileCheck.py
from os import listdir
from os.path import isfile, join


def FileCheck(filePath, dataBase):
    liste = []

    for i in listdir(filePath):
        if isfile(join(filePath, i)):
            liste.append(i)
        else:
            newPath = filePath + "/" + i
            FileCheck(newPath, dataBase)

    dataBase.append(liste)
